### ISSUE TITLE
run "run_once_install-packages.sh" again when another file changes

### ISSUE THREAD

**Issue by GabrielDuarteM** [twpayne/chezmoi#401]
I have a `run_once_install-packages.sh` file, and in there I run `brew bundle --global`, which tells brew to look to a file on the home called `.Brewfile` and installs all packages listed there. 
This `.Brewfile` is also on my chezmoi repo, but the thing is, if I change this file to add a new package, the `install-packages` script won't run again, as `install-packages` itself didn't change. 

Any idea how I can run it again when `.Brewfile` changes? I know I could probably create a new script without the `run_once_` suffix so it runs every time, but i'd like to only run it when needed, to make the applying process faster

**Comment by twpayne** [comment]
Some suggestions:

1. Have the script create `.Brewfile` instead of chezmoi, e.g. in your `run_once_install-packages`:

```sh
#!/bin/sh

cat > $HOME/.Brewfile <<EOF
brew "imagemagick"
brew "openssl"
EOF

brew bundle --global
```

2. Don't use `brew bundle`, instead install the packages explicitly in `run_once_install-packages`:

```sh
#!/bin/sh

brew install imagemagick
brew install openssl
```

3. Use a script that runs every time (not just once) and rely on `brew bundle` being idempotent.

4. Use a script that runs every time, records a checksum of `.Brewfile` in another file, and only runs `brew bundle --global` if the checksum has changed (and updates the recorded checksum after).

**Comment by GabrielDuarteM** [comment]
Ended up using the first option. Feels kinda like a hack tbh, but it works...

I was using option two before, but `brew install` fails with exit code 1 when the package is already installed, so I'd have to do something like `brew install openssl || true`, which would ignore all errors (including possibly errors that I should care about), so I opted to use the `.Brewfile`, which ignores the package when its already installed.

**Comment by karlwbrown** [comment]
I implemented #4 here:
https://gist.github.com/karlwbrown/7e48ebfdc3c14b3c879880d88bd77f66

Welcome feedback and improvements!
